### DISCUSSION TITLE
<WindowVirtualizer/> always scrolls to Top on Initial Scrolls

### DISCUSSION THREAD

**Discussion by xmsz-stu** [inokawa/virtua#580] [Q&A]
```
	<WindowVirtualizer
			onScroll={console.log}
			onScrollEnd={() => {
				console.log("end");
			}}
</WindowVirtualizer>
```

Scroll the page, it will  jumping back to the top during the first few scroll actions.

![局部截取_20241210_233109](https://github.com/user-attachments/assets/7104c0ff-6c16-47b7-b647-2404f7e26512)

**Comment by inokawa** [comment]
Hi, could you make a minimal reproduction?

**Reply by xmsz-stu** [reply]
I've found the problem. It's because my component has a lazy loading hook (I'm not sure about the exact reason). But when I wrap my component with a Suspense, the problem is solved and there won't be any issues.

**Reply by inokawa** [reply]
Thank you for the details.

Maybe you should use `window.scrollY` instead of `onScroll` prop of WindowVirtualizer. `onScroll` prop was newly added to `WindowVirtualizer` in `0.37.0` to deprecate `onRangeChange` prop. The correctness of the offset argument wasn't tested well.

**Reply by inokawa** [reply]
The argument of `onScroll` was omitted in `0.39.2` for now.
